superset/security/manager.py

```diff
@@ -221,6 +230,17 @@ class SupersetSecurityManager(  # pylint: disable=too-many-public-methods
         "all_query_access",
     )
 
+    guest_user_cls = GuestUser
+
+    def create_login_manager(self, app: Flask) -> LoginManager:
+        # pylint: disable=import-outside-toplevel
+        from superset.extensions import feature_flag_manager
+
+        lm = super().create_login_manager(app)
+        if feature_flag_manager.is_feature_enabled("EMBEDDED_SUPERSET"):
+            lm.request_loader(self.get_guest_user)
```

Comment:
I'm guessing this just registers another loader in the chain of "request_loaders"?

Reply:
There can only be one request loader set, no chain. But currently Superset doesn't set one. If there is a user session active, the loader doesn't get called, and if the loader returns None it's treated as if no session was found.

I know that in Shell we have our own request loader, and I've already made a branch where that is modified to be friendly with this change.